e2e-tests/controller-spark/controller_spark_sql_validation.sh

```diff
@@ -344,12 +347,60 @@ function validate_updated_resource() {
   fi
 }
 
+#################################################
+# Function that counts resources in  FHIR server and compares output to what is 
+#  in the source FHIR server
+# Globals:
+#   HOME_PATH
+#   SINK_FHIR_SERVER_URL
+#   TOTAL_TEST_PATIENTS
+#   TOTAL_TEST_ENCOUNTERS
+#   TOTAL_TEST_OBS
+#################################################
+function test_fhir_sink() {
```

Comment:
A good chunk of this function is very similar to `fhir_source_query` and both of these are similar to [this function in `pipeline_validation`](https://github.com/google/fhir-data-pipes/blob/168099505084b9ab72d51fb3558260bc557404bc/e2e-tests/pipeline_validation.sh#L148). Can you please refactor these into the same util function and reuse it everywhere? As a minimum please address the duplicate in this file but if you don't want to do the refactoring across files (since it is not introduced in this PR), that is fine but please file a clean-up issue for that.

Reply:
Thanks @bashir2 . I removed the duplicates in this file and created a separate i[ssue](https://github.com/google/fhir-data-pipes/issues/1065) to refactor this acros files